client/post-editor/term-token-field/index.jsx

```diff
@@ -96,8 +99,13 @@ TermTokenField.propTypes = {
 export default connect( ( state, props ) => {
 	const siteId = getSelectedSiteId( state );
 
+	const postId = getEditorPostId( state );
+	const postType = getEditedPostValue( state, siteId, postId, 'type' );
+	const taxonomy = getPostTypeTaxonomy( state, siteId, postType, props.taxonomyName );
+
 	return {
-		siteId: siteId,
+		siteId,
+		taxonomyLabel: taxonomy && taxonomy.label || props.taxonomyName,
```

Comment:
Should we be falling back to `props.taxonomyName` here? I imagine a `post_tag` heading would be a strange sight to see as an end-user. Might be better to omit the fallback altogether.